### ISSUE TITLE
on iOS on-screen keyboard covers multi-line input item, input is not visible

### ISSUE THREAD

**Issue by skrupenin** [beeware/toga#2672]
### Describe the bug

I'm building a chat application. Below is sub that generates main window.
On Android (physical device) and on iPhone 7 (simulator), on iPhone 7 (simulator) works well.

On iPhone 7, SE, 15 (physical devices), SE (simulator) when cursor put into multiline input (self.text_input) - virtual keyboard appears and covers input item. So user has no visibility on entered text.

    def create_conversation_ui(self):
        self.chat = toga.WebView(
            url=None,
            style=Pack(flex=1),
        )

        # Buttons
        self.text_input = toga.MultilineTextInput(
            style=Pack(flex=1, padding=5),
            placeholder='Type your',
            value = ''
        )
        self.send_button = toga.Button(
            "Send",
            on_press=self.handle_send_button,
            style=Pack(padding_top=5, padding_right=5),
        )
        share_feedback_button = toga.Button(
            "Share feedback",
            on_press=self.handle_share_feedback_button,
            style=Pack(padding_top=5, padding_right=5),
        )
        new_chat_button = toga.Button(
            "New chat",
            on_press=self.handle_new_chat_button,
            style=Pack(padding_top=5, padding_right=5),
        )

        buttons_box = toga.Box(
            children=[self.send_button,
                      share_feedback_button,
                      new_chat_button,
                      toga.Label('id: ' + device_id, style=Pack(color=GRAY))],
            style=Pack(direction=COLUMN),
        )

        input_box = toga.Box(
            children=[self.text_input, buttons_box],
            style=Pack(direction=ROW),
        )

        # Outermost box
        outer_box = toga.Box(
            children=[self.chat, input_box],
            style=Pack(direction=COLUMN),
        )

        return outer_box

### Steps to reproduce

1. Build interface and run on iPhone 7, SE, 15
2. Go to 'Type your' multiline input text
3. See the issue

### Expected behavior

When cursor placed into multiline input and virtual keyboard displayed, application window either scrolls up or webview piece decreased in height so that user can wee both multiline text input and virtual keyboard. 

### Screenshots

![IMG_9893](https://github.com/beeware/briefcase/assets/6624262/9bcd5ce8-1201-4597-be6b-906fcd71a050)
![IMG_9894](https://github.com/beeware/briefcase/assets/6624262/69621530-0bcc-4fcf-a204-8573dd35d7a2)
![IMG_9895](https://github.com/beeware/briefcase/assets/6624262/9087d7ba-4add-4ddc-a473-94875e8bb97f)
![IMG_9896](https://github.com/beeware/briefcase/assets/6624262/c246470e-a5be-4c6b-b9b5-33fa5fa3328b)

### Environment

- Operating System: iOS 17.5.1
- Python version: 3.12
- Software versions:
  - Briefcase: 0.3.19
  - Toga: 0.4.5

**Comment by freakboy3742** [comment]
Moving to the Toga repo, as this isn't an issue with how Briefcase deploys apps, it's an issue with the Toga widget kit.

I think I understand the problem you're describing; however, it's not immediately obvious to me what the "right" behavior should be. 

Resizing the app's window to accomodate the keyboard would be the simplest approach; but that would also have a significant impact on how apps render. It would be easy to end up with a situation where the app is unable to be rendered when the keyboard is displayed, because the minimum rendering size can't be accommodated in the new, smaller size. 

As an aside, when reporting a problem, the most helpful thing you can do is provide a *minimal, complete example* - a *full* app (not just a block of code that might contain the problem), that *only* has the pieces necessary to reproduce the problem. If you're going to provide screenshots, provide screenshots of *that* app - not screenshots that need substantial redaction an in-image explanatory notes for what the image isn't showing.

**Comment by mhsmith** [comment]
Similar issue on Android: #2634

**Comment by johnzhou721** [comment]
Resizing application window strikes me as a wrong approach.  Take the new iOS 26 messages app --  on the bottom when the keyboard pops up, it actually actively blurs the non-scrolled-up content in the scroll view.

The most common usecase for this is like the bottom input of a chat window like this -- and it seems that the Messages app has a kind of bottom bar (this is macOS UI but iOS is similar) that automatically pops up when you press on the keyboard:

<img width="518" height="996" alt="Image" src="https://github.com/user-attachments/assets/68b7f251-918d-4698-bc11-f0d83ea0e647" />

So I propose the following:
- Make ScrollContainer able to scroll its bottom above the keyboard when it pops up.  This can be done by adjusting contentInset by listening to when the keyboard pops up.
- Investigate a way to add a "bottom bar that will pop up" structure to a ScrollContainer.

---

EDIT -- if this is acceptable, then I will start to fix this after 3 or 4 months when I finish churning the iOS layout stuff a lot.

**Comment by freakboy3742** [comment]
The contentInset approach might be worthwhile as a fix in itself - but the problem needs to have a solution when scrolling isn't involved. The problem is that the main window content isn't *required* to be scrollable; if the keyboard is displayed, that means any control under the keyboard is unpressable.

**Comment by johnzhou721** [comment]
@freakboy3742 yeah — bottom controls and scroll inset handling is a common pattern though, so getting that done will take lots of pressure off a general solution.
